src/components/AppLayout/Header/index.tsx

```diff
@@ -54,7 +56,11 @@ const HeaderComponent = (): React.ReactElement => {
 
   const getProviderDetailsBased = () => {
     if (!loaded) {
-      return <ConnectDetails />
+      return (
+        <Suspense fallback={null}>
```

Comment:
I tried using something other than `null` as fallback but it doesn't make any difference since the component is hidden. I would keep null here and use `wrapInSuspense` like we do in other places.

Reply:
Is there a visible loading period when using `null`?

Reply:
Not that I can tell. The component is immediately visible as soon as the user clicks on "Connect Wallet".